src/cheri/xycheriot.adoc

```diff
@@ -0,0 +1,511 @@
+
+No <<sec_cap_type>> value is considered
+<<sec_cap_type_ambient,ambiently available>>,
+even those used by <<section_rvy_sentry_jalr_ext>> (see below).
+The <<section_rvy_sentry_insn_ext>> extension (and its <<SENTRY>> instruction)
+is ambient.
+If the result of a <<CBLD>> instruction has a set {ctag},
+then it will also be unsealed.
+
+=== Sentries
+
+{cheriot_ext_name} builds upon <<section_rvy_sentry_jalr_ext>>
+to give architectural semantics to _five_ <<sec_cap_type>> values
+({cheriot_sentry_ot_min} through {cheriot_sentry_ot_max}, inclusive):
+
+* Any of these values may be passed as inputs to <<JALR_CHERI>>.
+  When such a <<JALR_CHERI>> instruction retires,
+  the hart's `mstatus.MIE` bit is updated as per <<tab_cheriot_sentry_irq>>.
+
+* Two values, {cheriot_sentry_ot_bd} and {cheriot_sentry_ot_be},
+  are used by <<JALR_CHERI>> to seal the return capability.
+  If the hart's `mstatus.MIE` bit is `0` when the instruction retires,
+  the return capability is sealed with {cheriot_sentry_ot_bd};
+  otherwise, {cheriot_sentry_ot_be} is used.
+
+* <<JALR_CHERI>> is given conditional behavior
+  based on the register selectors used,
+  as detailed in <<tab_cheriot_jalr_regs>>.
+  Prohibited combinations of register selectors and <<sec_cap_type>> value
+  will raise a CHERI exception.
+
+[NOTE]
+=====
+All of the <<sec_cap_type>> values used with <<section_rvy_sentry_jalr_ext>>
+are encodable if and only if the capability grants <<x_perm>>.
+The remaining two <<x_perm>>-associated <<sec_cap_type>> values
+remain available for software's use,
+as do all of the <<sec_cap_type>> values
+associated with capabilities not granting <<x_perm>>.
+=====
+
+.Additional <<JALR_CHERI>> Architectural Semantics
+[#tab_cheriot_sentry_irq,width="100%",options=header,halign=center,cols="1,5"]
+|===
+| <<sec_cap_type>>       | IRQs at Retirement (`mstatus.MIE`)
+| {cheriot_sentry_ot_fi} | Unchanged (as is)
+| {cheriot_sentry_ot_fd}, {cheriot_sentry_ot_bd} | Deferred (`0`)
+| {cheriot_sentry_ot_fe}, {cheriot_sentry_ot_be} | Enabled (`1`)
+|===
+
+.<<JALR_CHERI>> Conditional Behavior
+[#tab_cheriot_jalr_regs,width="100%",options=header,halign=center,cols="1,2,4,4"]
+|===
+| {cs1} | {cd}   | Permitted {cs1} <<sec_cap_type>>-s | Comments
+| `ra`  | `null` | {cheriot_sentry_ot_bd}, {cheriot_sentry_ot_be} | Function return
+| ≠`ra` | `null` | 0, {cheriot_sentry_ot_fi} | Tail call
+| any   | ∉{ `null`, `ra` } | 0, {cheriot_sentry_ot_fi} | Code outlining
+| any   | `ra`   | 0, {cheriot_sentry_ot_fi}, {cheriot_sentry_ot_fd}, {cheriot_sentry_ot_fe} | Function call
+|===
+
+[NOTE]
+=====
+TODO: Research: can we refine that table to introduce "must be unsealed" and
+"must be sealed" forward arc rows, or do we need a second instruction instead?
+
+How hard is it to explain such a thing to LLVM?  Would we need to push type
+refinements up onto C / C++ function pointers?
+
+=====
+
+=== Miscellaneous Changes
+
+[NOTE]
+=====
+TODO: What else am I forgetting?
```

Comment:
I believe you have the bounds checking on branch rather than at target? Or will that be removed from CHERIoTv2?

Reply:
I added a paragraph noting that we might allow-but-not-mandate it.  We'll see how the uarches evolve. :)